x/liquidstakeibc/migrations/stkosmo/store.go

```diff
@@ -0,0 +1,143 @@
+package stkosmo
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	"github.com/persistenceOne/pstake-native/v2/x/liquidstakeibc/types"
+)
+
+const (
+	OsmosisTestnetChainID   = "osmo-test-5"
+	OsmosisTestnetMintDenom = "stk/uosmo"
+)
+
+// MigrateStore performs in-place store migrations from v2.3.0 to remove stkOSMO from test-core-2.
+// The migration includes:
+//
+// - Burn all minted stkOSMO.
+// - Remove all the related store objects.
+// - Migrate stkOSMO host chain to remove it from the store.
+func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec, k types.BankKeeper) error {
+
+	// Burn all the coins present in the module account.
+	if err := k.BurnCoins(
+		ctx,
+		types.ModuleName,
+		sdk.NewCoins(k.GetBalance(ctx, authtypes.NewModuleAddress(types.ModuleName), OsmosisTestnetMintDenom)),
+	); err != nil {
+		return err
+	}
+
+	// Burn all coins on other addresses.
+	k.IterateAllBalances(
+		ctx,
+		func(address sdk.AccAddress, coin sdk.Coin) (stop bool) {
+
+			if coin.Denom == OsmosisTestnetMintDenom {
+				// Send the whole address balance to the module account.
+				if err := k.SendCoinsFromAccountToModule(ctx, address, types.ModuleName, sdk.NewCoins(coin)); err != nil {
```

Comment:
can you check if you are allowed to update balances while you are iterating through the same store. There should be a lock on it while you are iterating

Reply:
I just tested the solution and amounts are being burned, so I guess there is no issue there, the lock might be while the actual iteration is being performed.